vcsclient/fs.go

```diff
@@ -200,6 +197,28 @@ func GetFileWithOptions(fs vfs.FileSystem, path string, opt GetFileOptions) (*Fi
 	return &fwr, nil
 }
 
+// readDir uses the passed vfs.FileSystem to read from starting at the base path. If
+// shouldRecurse is set to true, it will recurse into all sub-folders and return the
+// full sub-tree.
+func readDir(fs vfs.FileSystem, base string, shouldRecurse bool) ([]*TreeEntry, error) {
```

Comment:
Well it's something as simple as:
```
type walkableFS { vfs.FileSystem }

func (w walkableFS) Join(p ...string) string { return filepath.Join(p...) }
```
and then `WalkFS(path, walkableFS{fs})` but this doesn't create a tree so I think my solution is more appropriate.